src/Mvc/Mvc.ViewFeatures/src/Rendering/CheckBoxHiddenInputRenderMode.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.Rendering
+{
+    /// <summary>
+    /// Controls the rendering of hidden input fields when using CheckBox tag helpers or html helpers.
+    /// </summary>
+    public enum CheckBoxHiddenInputRenderMode
+    {
+        /// <summary>
+        /// Hidden input fields will not be automatically rendered. If checkbox is not checked, no value will be posted.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Hidden input fields will be rendered inline with each checkbox. Use this for legacy ASP.NET MVC behavior.
+        /// </summary>
+        Inline = 1,
+
+        /// <summary>
+        /// Hidden input fields will be rendered for each checkbox at the bottom of the form element. Use this for default MVC behavior.
```

Comment:
Comment should be more specific. Mention rendering at the end of the form element is _preferred_ but the hidden input field will be rendered inline with the checkbox if `!ViewContext.FormContext.CanRenderAtEndOfForm`.

Reply:
Also, please make sure tests cover this fallback to `Inline` behaviour for both HTML and tag helpers. (The code looks correct but test to keep it that way.)